### PR TITLE
feat: add zrx proxy

### PR DESCRIPTION
**Pull request by kaladinlight**
- Supports both v1 and v2 apis.
  - v1 leverages chain path for correct routing
  - v2 leverages `chainId` query param for correct routing
- No backend response cache at this time

### CODE REVIEW

**Comment by kaladinlight** [comment]
closes: https://github.com/shapeshift/web/issues/7784

File: node/proxy/api/src/zrx.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  constructor() {
+    this.axiosInstance = axios.create({
+      headers: {
+        '0x-api-key': ZRX_API_KEY,
+      },
+    })
+  }
+
+  async handler(req: Request, res: Response): Promise<void> {
+    const parsedUrl = new URL('https://dummy.com'.concat(req.url))
```
Comment (gomesalexandre) [inline node/proxy/api/src/zrx.ts:21]: Is there any rationale here re: not using req.url directly?

File: node/proxy/api/src/zrx.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    const path = parsedUrl.pathname.substring('/api/v1/zrx/'.length)
```
Comment (gomesalexandre) [inline node/proxy/api/src/zrx.ts:22]: 🧠

Comment (gomesalexandre) [reply node/proxy/api/src/zrx.ts:22]: Could potentially use a capture group to make this less leetcody magic but meh

File: node/proxy/api/src/zrx.ts (hunk 1 above)
Comment (kaladinlight) [reply node/proxy/api/src/zrx.ts:21]: `req.url` doesn't include the base url and will not be parsed correctly by `URL`, so just added a dummy base url so we can use it and make sure there is no funny business with pulling off the query params (versus just split on `?`)

File: node/proxy/api/src/zrx.ts (hunk 2 above)
Comment (kaladinlight) [reply node/proxy/api/src/zrx.ts:22]: was following the pattern, though not sure how I ended up with substring at one point lol. I will just update with a simple `.replace()`

**Pull request merged**